src/Illuminate/Database/Eloquent/Concerns/GuardsAttributes.php

```diff
@@ -151,8 +149,7 @@ public function isFillable($key)
             return false;
         }
 
-        return empty($this->getFillable()) &&
-            ! Str::startsWith($key, '_');
+        return empty($this->getFillable()) && $key[0] !== '_';
```

Comment:
Will fatal error if the key is the empty string.

Reply:
Right, please check the new commit.

Reply:
@GrahamCampbell or do you think `(string)` casting would be better?